### PR TITLE
Issue-5152: Old Assessments/directories throwing errors

### PR DESCRIPTION
**Pull request by rhernandez-intertech**
connects #5152 #5103

### CODE REVIEW

File: src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts [hunk 1]
```diff
@@ -40,7 +40,9 @@ export class DirectoryItemComponent implements OnInit {
     this.directory.selected = false;
     this.updateDashboardDataSub = this.dashboardService.updateDashboardData.subscribe(val => {
       this.directory = this.directoryDbService.getById(this.directory.id);
-      this.directory.selected = false;
+      if(this.directory){
```
Comment (rmroot) [inline src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts:43]: the check for directory doesn't really do anything here. The line previous uses `this.directory.id` and a couple lines earlier has `this.directory.selected`

File: src/app/dashboard/directory-dashboard/directory-items/pre-assessment-item/pre-assessment-item.component.ts [hunk 2]
```diff
@@ -66,10 +66,14 @@ export class PreAssessmentItemComponent implements OnInit {
...
-    let tmpResults = this.preAssessmentService.getResults(this.calculator.preAssessments, this.settings, 'MMBtu', false);
-    this.energyUsed = _.sumBy(tmpResults, 'value');
-    this.energyCost = _.sumBy(tmpResults, 'energyCost');
+    if (this.calculator.preAssessments) {
+      this.numUnits = this.calculator.preAssessments.length;
+      let tmpResults = this.preAssessmentService.getResults(this.calculator.preAssessments, this.settings, 'MMBtu', false);
+      this.energyUsed = _.sumBy(tmpResults, 'value');
+      this.energyCost = _.sumBy(tmpResults, 'energyCost');
+    } else {
+      this.calculator.type = 'pump';
```
Comment (rmroot) [inline src/app/dashboard/directory-dashboard/directory-items/pre-assessment-item/pre-assessment-item.component.ts:75]: I'm not sure what setting the type to "pump" is accomplishing

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts [hunk 3]
```diff
@@ -39,8 +39,7 @@ export class DirectoryItemComponent implements OnInit {
   ngOnInit() {    
     this.directory.selected = false;
     this.updateDashboardDataSub = this.dashboardService.updateDashboardData.subscribe(val => {
-      this.directory = this.directoryDbService.getById(this.directory.id);
-      this.directory.selected = false;
```
Comment (rmroot) [inline src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts:43]: we will still want this selected value set to false when this behavior subject is fired

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts (hunk 3 above)
Comment (rhernandez-intertech) [reply src/app/dashboard/directory-dashboard/directory-items/directory-item/directory-item.component.ts:43]: in order to have the `this.directory.selected = false;` it needs to be in an If statement checking if that directory exists or else an error gets thrown

**Review by rmroot — APPROVED** [review]

**Pull request merged**